lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -166,19 +228,23 @@ def reset!
         super
       end
 
-      def enabled?
+      def logger_enabled?
         @enabled && @instrumentation_logger_enabled
       end
 
+      def logstasher_enabled?
+        @enabled && NewRelic::Agent::Instrumentation::LogStasher.enabled?
```

Comment:
If you're following the existing `Logger` pattern for `LogStasher`, perhaps you should have a dedicated `@instrumentation_logstasher_enabled` instance var set at initialize time to match `Logger`.

Reply:
This [commit](https://github.com/newrelic/newrelic-ruby-agent/pull/2559/commits/02d09b3fba028b4dab5a46edc6e95cdce2d84eee) matches Logger. Load order was an issue, so we aren't able to use the dynamic `NewRelic::Agent.base_name(name)` and need to require LogStasher instrumentation early on. I'm leaning more towards not having a LogStasher instance var set at initialize so that we add less code that may have to be adjusted when more loggers are added. But no strong opinion either way

Reply:
I've reverted this change for now!